src/overlay/Peer.cpp

```diff
@@ -1029,12 +1063,61 @@ Peer::addMsgAndMaybeTrimQueue(std::shared_ptr<StellarMessage const> msg)
             }
         }
     }
+    else if (type == FLOOD_ADVERT)
+    {
+        uint32_t limit = mApp.getLedgerManager().getLastMaxTxSetSizeOps() /
+                         mApp.getConfig().FLOOD_MAX_ADVERT_SIZE;
+        if (limit < 1)
+        {
+            limit = 1;
+        }
+        if (queue.size() > limit)
+        {
+            dropped = queue.size() - limit;
+            queue.erase(queue.begin(), queue.begin() + dropped);
+            TracyPlot("overlay-adverts-dropped", static_cast<int64_t>(dropped));
+        }
+    }
+    else if (type == FLOOD_DEMAND)
+    {
+        uint32_t limit = mApp.getLedgerManager().getLastMaxTxSetSizeOps() /
```

Comment:
same thing here: we might not have demands on max size

Reply:
Fixed.